src/WinRT.Runtime/ComWrappersSupport.cs

```diff
@@ -281,21 +287,36 @@ private static bool IsIReferenceArray(Type implementationType)
             return Expression.Lambda<Func<IInspectable, object>>(
                 Expression.Call(type.GetHelperType().GetMethod("CreateRcw", BindingFlags.Public | BindingFlags.Static), 
                     parms), parms).Compile();
+        }
+
+        internal static Func<IntPtr, object> CreateDelegateFactory(Type type)
+        {
+            return DelegateFactoryCache.GetOrAdd(type, (type) =>

```

Comment:
It could be, but there isn't too many types in WinRT.Runtime which can be added here.  Most are in the generated projections.  There looks to be only 3 of which only like 2 can be added here.  Right now looking at traces, this function doesn't seem to be expensive, but if either of those 2 do end up being expensive, we can add them in the future.